### PR TITLE
Add tests for form_text_field.1.dart

### PR DESCRIPTION
**Pull request by ValentinVignal**
Contributes to https://github.com/flutter/flutter/issues/130459

It adds a test for
- `examples/api/lib/material/text_form_field/text_form_field.1.dart`

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] I followed the [breaking change policy] and added [Data Driven Fixes] where supported.
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[Data Driven Fixes]: https://github.com/flutter/flutter/wiki/Data-driven-Fixes

### CODE REVIEW

File: examples/api/test/material/text_form_field/text_form_field.1_test.dart [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    expect(textFormField, findsExactly(5));
+
+    final Finder editableText = find.byType(EditableText);
+    expect(editableText, findsExactly(5));
+
+    List<bool> getFocuses() {
+      return editableText.evaluate()
+        .map((Element finderResult) => (finderResult.widget as EditableText).focusNode.hasFocus)
+        .toList();
+    }
```
Comment (ValentinVignal) [inline examples/api/test/material/text_form_field/text_form_field.1_test.dart:24]: Do you have a better way to get the focuses?

Comment (bleroux) [reply examples/api/test/material/text_form_field/text_form_field.1_test.dart:24]: Nothing simpler in mind, I think this utility function is readable as-is 👍

File: examples/api/test/material/text_form_field/text_form_field.1_test.dart [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    await tester.pump();
+
+    expect(getFocuses(), const <bool>[false, false, false, false, true]);
+
+    await tester.sendKeyEvent(LogicalKeyboardKey.space);
+    await tester.pump();
+
+    expect(getFocuses(), const <bool>[true, false, false, false, false]);
+  });
+
```
Comment (bleroux) [inline examples/api/test/material/text_form_field/text_form_field.1_test.dart:58]: Consider removing this extra line

**Review by bleroux — APPROVED** [review]
LGTM!

**Review by TahaTesser — APPROVED** [review]

**Comment by cbracken** [comment]
Reason for revert: Surprisingly, the following test seems to be consistently failing on Windows after the addition of this test:
```
flutter/packages/flutter/test/widgets/sliver_tree_test.dart: .toggleNodeWith, onNodeToggle
```

**Comment by cbracken** [comment]
Reverted this since the above-mentioned test was failing consistently starting at this patch. The main possibility I was envisioning was some sort of non-hermetic crossover between the tests, but even after the revert, the above-mentioned test is consistently failing. It seems there's some other non-hermetic component of our tests.

@ValentinVignal apologies for the noise here. Please re-send out this patch and we'll get it re-landed.

**Comment by ValentinVignal** [comment]
Sure @cbracken , no problem, I re-created the PR here: https://github.com/flutter/flutter/pull/150750

**Comment by cbracken** [comment]
Thank @ValentinVignal and apologies again for the churn!

**Pull request merged**